src/cargo/core/resolver/mod.rs

```diff
@@ -590,6 +591,15 @@ impl DepsFrame {
             .map(|(_, (_, candidates, _))| candidates.len())
             .unwrap_or(0)
     }
+
+    fn flatten<'s>(&'s self) -> Box<Iterator<Item = (PackageId, Dependency)> + 's> {
+        // TODO: with impl Trait the Box can be removed
+        Box::new(
+            self.remaining_siblings
+                .clone()
+                .map(move |(_, (d, _, _))| (self.parent.package_id().clone(), d)),
```

Comment:
If we worry about perf at some point I think the `clone()` here could be deferred to a later point probably

Reply:
It is an `Arc` clone, and I could not convinced the borrow checker that it lived long enough without it. 
edit: now it works. good call.